### ISSUE TITLE
file:../.. not supported

### ISSUE THREAD

**Issue by staltz** [pnpm/pnpm#772]
If you have a project structure like

```
root
  package.json
  src
  examples
    B
      package.json
```

where B/package.json depends on `file:../..`, in other words root/package.json, then it appears pnpm doesn't correctly install it and crashes.

### pnpm version:

0.68.0

### Code to reproduce the issue:

Git clone this https://github.com/staltz/cycle-onionify/tree/27edc145594cc3bc890d10639354a054c6b2fe37/examples/lenses

```
git clone git@github.com:staltz/cycle-onionify.git
```

```
cd cycle-onionify/examples/lenses
```

```
pnpm i
```

### Expected behavior:

pnpm install node_modules

### Actual behavior:

Error on `pnpm i`:

```
  browserify 13.0.x                                        linking 13.0.1 ↓
  mkdirp 0.5.x                                              linking 0.5.1 ↓
  typescript 2.3.2                                          linking 2.3.2 ↓
  @cycle/run 3.0                                            linking 3.0.0 ↓
  @cycle/dom 15.2                          installing dependencies 15.2.0 .
  @cycle/isolate 2.1.0                                      linking 2.1.0 ↓
  @types/es6-shim ^0.31.32                                linking 0.31.33 ↓
  cycle-onionify ../..                                         installing .
  xstream 10.3                             installing dependencies 10.3.0 .
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open '/Users/staltz/onionify/> cycle-onio
nify@3.3.0 prepublish /Users/staltz/onionify
> npm run lib

> cycle-onionify@3.3.0 prelib /Users/staltz/onionify
> mkdir -p lib

> cycle-onionify@3.3.0 lib /Users/staltz/onionify
> tsc

cycle-onionify-3.3.0.tgz'
    at Error (native)
```

Also, it created a `.tgz` file in the root.

### Additional information:

 - `node -v` prints: v6.10.1
 - Windows, OS X, or Linux?: macOS

**Comment by zkochan** [comment]
npm5 currently symlinks packages specified via `file:..` (see https://github.com/npm/npm/pull/15900). We'll have to do probably the same

**Comment by donaldpipowitch** [comment]
Just wanted to write a small test case as a follow up to this [Twitter thread](https://twitter.com/ZoltanKochan/status/875720816385830912), but I get an error. (Don't know why... I used `pnpm` and `file:` earlier today without this error in a bigger project.)

---


### pnpm version:

0.70.0

### Code to reproduce the issue:

`./package.json`:

```
{
  "name": "foo",
  "version": "1.0.0"
}
```

`./examples/package.json`:

```
{
  "name": "bar",
  "version": "1.0.0",
  "dependencies": {
    "foo": "file:../"
  }
}
```

```bash
$ cd examples && pnpm install
  foo file:../                                                      1.0.0 ✓
  WARN Cannot find resolution of /foo/file:.. in shrinkwrap file
```

However it looks like everything is installed correctly.

**Expected behavior:**

No warning? Or a different warning? Currently I don't know what is the problem.

---

Now I add a new file `./server.js` with the content `console.log('foo');` and call `$  pnpm install` again inside `examples/`, but no `./examples/node_modules/foo/server.js` appears. (Even when I remove `./examples/node_modules` and `./examples/shrinkwrap.yaml` and try it again.)

**Expected behavior:**

`./examples/node_modules/foo/server.js` should exist.

---

Lol. I just called `$ pnpm install foo` inside `examples/` and [`foo`](https://www.npmjs.com/package/foo) was installed and in the `./examples/package.json` the line `"foo": "file:../"` changed to `"foo": "^1.0.0"`. 😄 

**Expected behavior:**

Just install `./` (the local `foo` package) again and don't change the content of `./examples/package.json`. (This is the behaviour of `$ npm install foo`.)

---

_Thank you_ for `pnpm`! ❤️

**Comment by zkochan** [comment]
I think currently the easiest way to solve this is running `pnpm link` for the `file:` dependencies. So pnpm will skip those dependencies during installation and just run `pnpm link ../whatever` afterward.

This might be not the fastest solution from performance point of view, but it would introduce less complexity to the system which is currently complex as hell already :-D

**Comment by zkochan** [comment]
@donaldpipowitch first part of your issue is going to be fixed by the #807.

For the second part about the rewritten spec in `package.json` I created a separate issue: #809

**Comment by donaldpipowitch** [comment]
This is awesome, thank you. So this basically behaves like `nom@5` now, right?

I probably will comment about this behavior in mom's issue on this, too, because there is one downside about the symlink. They don't behave like a normally published module, because everything is symlink even if you use a `.gitignore`, `.npmignore` or `"files"`. But I guess this is a bigger problem...

**Comment by zkochan** [comment]
This issue is closed by [0.71.0](https://github.com/pnpm/pnpm/releases/tag/v0.71.0)

@donaldpipowitch yes. This is a bigger problem. I was thinking about this as well. Maybe the project can be structured in a way that would allow avoiding usage of `.npmignore`.

**Comment by donaldpipowitch** [comment]
Cool. Thank you!
